apps/core/app/api/product/[id]/route.ts

```diff
@@ -1,8 +1,32 @@
+/*
+ * This route handler is exclusively for use by the product sheet component.
+ */
 import { NextRequest, NextResponse } from 'next/server';
 
-import { getProduct } from '~/client/queries/get-product';
+import { client } from '~/client';
+import { graphql } from '~/client/graphql';
+import { ProductSheetContentFragment } from '~/components/product-sheet/fragment';
+
+const GetProductQuery = graphql(
+  `
+    query GetProductQuery($productId: Int!, $optionValueIds: [OptionValueId!]) {
+      site {
+        product(entityId: $productId, optionValueIds: $optionValueIds) {
+          ...ProductSheetContentFragment
+        }
+      }
+    }
+  `,
+  [ProductSheetContentFragment],
+);
 
 export const GET = async (request: NextRequest, { params }: { params: { id: string } }) => {
+  if (request.headers.get('x-catalyst-product-sheet') !== 'true') {
+    return NextResponse.json('Endpoint only available for product-sheet component.', {
```

Comment:
🍹 Do you think we should normalize this error as the same as what @jorgemoya has been doing with server actions?, e.g. `{ status: 'error', message: '...', }`

Reply:
Added in https://github.com/bigcommerce/catalyst/compare/bbe1ff9f93627d74585a2ac3d40ecc226ead6c1e..3d80524391e2c26916714c6ce4ca398bd61ba57a